pyshelf/link_manager.py

```diff
@@ -9,21 +9,45 @@ def __init__(self, container):
         self.request = self.container.request
         self.path_builder = self.container.artifact_path_builder
 
-    def assign_listing(self, artifact_list):
+    def assign_listing_path(self, artifact_path_list):
```

Comment:
So after this was added I think it seems clear that the responsibility of this manager should be to deal with paths... ya think?  The artifact preprocessing can be done to generate a list of paths (or a single path) and then it can be provided to this manager